python-backend/app/api/mines/variances/resources/variance.py

```diff
@@ -0,0 +1,237 @@
+    )
+    @requires_any_of([MINE_VIEW])
+    @api.marshal_with(variance_model, code=200)
+    def get(self, mine_guid, variance_id):
+        variance = Variance.find_by_variance_id(variance_id)
+
+        if variance is None:
+            raise BadRequest('Unable to fetch variance')
+
+        return variance
+
+
+class VarianceDocumentUploadResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('mine_document_guid', type=str)
+    parser.add_argument('document_manager_guid', type=str)
+    parser.add_argument('document_name', type=str)
+
+
+    @api.doc(
+        description=
+        'Request a document_manager_guid for uploading a document'
+    )
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def post(self, mine_guid, variance_id):
+        data = self.parser.parse_args()
+        metadata = self._parse_request_metadata()
+        if not metadata or not metadata.get('filename'):
+            raise BadRequest('Filename not found in request metadata header')
+
+        # Save file
+        mine = Mine.find_by_mine_guid(mine_guid)
+        document_name = metadata.get('filename')
+        data = {
+            'folder': f'mines/{mine.mine_guid}/variances',
+            'pretty_folder': f'mines/{mine.mine_no}/variances',
+            'filename': document_name
+        }
+        document_manager_URL = f'{current_app.config["DOCUMENT_MANAGER_URL"]}/document-manager'
+
+        resp = requests.post(
+            url=document_manager_URL,
+            headers={key: value
+                     for (key, value) in request.headers if key != 'Host'},
+            data=data,
+            cookies=request.cookies,
+        )
+
+        response = Response(str(resp.content), resp.status_code, resp.raw.headers.items())
+        return response
+
+    @api.doc(
+        description=
+        'Associate an uploaded file with a variance.',
+        params={
+            'mine_guid': 'guid for the mine with which the variance is associated',
+            'variance_id': 'ID for the variance to which the document should be associated'
+        }
+    )
+    @api.marshal_with(variance_model, code=200)
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def put(self, mine_guid, variance_id):
+        variance = Variance.find_by_variance_id(variance_id)
+
+        data = self.parser.parse_args()
+        document_name = data.get('document_name')
+        document_manager_guid = data.get('document_manager_guid')
+
+        if not document_name:
+            raise BadRequest('Missing document_name')
+
+        # Register new file upload
+        mine_doc = MineDocument(
+            mine_guid=mine_guid,
+            document_manager_guid=document_manager_guid,
+            document_name=document_name)
+
+        if not mine_doc:
+            raise BadRequest('Unable to register uploaded file as document')
+
+        variance.documents.append(mine_doc)
+        db.session.commit()
```

Comment:
variance.save() should do the same thing but use the try/except we in that method.